streaming/receiver.c

```diff
@@ -418,17 +418,20 @@ static int rrdpush_receive(struct receiver_state *rpt)
     log_stream_connection(rpt->client_ip, rpt->client_port, rpt->key, rpt->host->machine_guid, rpt->host->hostname, "DISCONNECTED");
```

Comment:
Still crashing in line 418:
```
(gdb) list *0x560535
0x560535 is in rrdpush_receive (streaming/receiver.c:418).
413
414
415         size_t count = streaming_parser(rpt, &cd, fp);
416         //size_t count = pluginsd_process(host, &cd, fp, 1);
417
418         log_stream_connection(rpt->client_ip, rpt->client_port, rpt->key, rpt->host->machine_guid, rpt->host->hostname, "DISCONNECTED");
419         error("STREAM %s [receive from [%s]:%s]: disconnected (completed %zu updates).", rpt->host->hostname, rpt->client_ip, rpt->client_port, count);
420
421         // During a shutdown there is cleanup code in rrdhost that will cancel the sender thread
422         if (!netdata_exit) {
(gdb) 
```
You should include those lines in the conditional as well.

Reply:
Facepalm. I'll move them inside. I couldn't trigger it in testing though - went up to 25 child nodes. Are you doing anything other than shutting down the parent while the child nodes are streaming data?